sensor/common/registry/tlscheck_test.go

```diff
@@ -0,0 +1,55 @@
+package registry
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCheckTLS(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("secure", func(t *testing.T) {
+		regStore := NewRegistryStore(alwaysSecureCheckTLS)
+		secure, skip, err := regStore.checkTLS(ctx, "fake")
+		assert.True(t, secure)
+		assert.False(t, skip)
+		assert.NoError(t, err)
+
+		// Ensure the results do not change when attempted again / using cache
```

Comment:
up to you, but maybe we can add a check for the length of the cache to confirm this is cached

Reply:
https://github.com/stackrox/stackrox/pull/7962/commits/b1a79ca548cb561ef895aa3fbaf518b30a28b4d9